storage/remote/codec.go

```diff
@@ -898,28 +899,48 @@ func DecodeOTLPWriteRequest(r *http.Request) (pmetricotlp.ExportRequest, error)
 		return pmetricotlp.NewExportRequest(), fmt.Errorf("unsupported content type: %s, supported: [%s, %s]", contentType, jsonContentType, pbContentType)
 	}
 
-	reader := r.Body
+	var reader io.Reader = r.Body
+	var gzipReader *gzip.Reader
+	var zstdDecoder *zstd.Decoder
 	// Handle compression.
 	switch r.Header.Get("Content-Encoding") {
 	case "gzip":
-		gr, err := gzip.NewReader(reader)
+		gr, err := gzip.NewReader(r.Body)
 		if err != nil {
 			return pmetricotlp.NewExportRequest(), err
 		}
 		reader = gr
-
+		gzipReader = gr
+	case "zstd":
+		zd, err := zstd.NewReader(r.Body)
+		if err != nil {
+			return pmetricotlp.NewExportRequest(), err
+		}
+		reader = zd
+		zstdDecoder = zd
```

Comment:
similar, deferring here is possible, avoiding boilerplate code below with extra variables.

```suggestion
		defer zd.Close()
```